### PR TITLE
Add JS docstrings

### LINKED ISSUE #293: Fix JS warnings
The current JS linter uses `--jscomp_warning` and doesn't fail on (many) warnings. I'd like to change this to `--jscomp_error`.

### PR DESCRIPTION
**Pull request by connorcoley**
Fixes #293

### CODE REVIEW

**Comment by connorcoley** [comment]
I would appreciate any input from @skearnes @antonkast-google or @n8kim1 about these docstrings. If you like them, I'd suggest that the load/unload descriptions can probably be used as a template for the other files. Currently, I've only updated `amounts.js`, `amountCrudes.js`, and `compounds.js` so they pass the linter.

P.S. I'm following https://google.github.io/styleguide/jsguide.html#jsdoc

**Comment by skearnes** [comment]
Thanks Connor, this looks great! I'll start from the bottom of the directory and work my way up. We can meet somewhere near the middle (and Nathan, if you join in maybe start at inputs.js or nodes.js?)

**Comment by skearnes** [comment]
FYI just pushed workups.js to make sure everything works with my setup; I was slightly less verbose than Connor :)

**Comment by antonkast-google** [comment]
It's great to document all this. I should have been doing that myself.

A couple suggestions based on personal experience:

1) Comments should add information, e.g. avoid "setValue() /* Sets the value. */".

2) Type annotations are great so long as there is enforcement. Otherwise, comments are almost always read from source, so leaving out @param etc. is actually more comprehensible.

**Comment by skearnes** [comment]
Note that `@param` is still useful even without type annotations as structured documentation.

Do you know how to adjust the compilation to enforce types at compile- and run-time? We should do this if we can.

**Comment by skearnes** [comment]
@connorcoley note that `@return` does not need a variable name; https://google.github.io/styleguide/jsguide.html#jsdoc-method-and-function-comments

**Comment by connorcoley** [comment]
Right, that makes sense. Thanks

**Comment by skearnes** [comment]
Good to go! @antonkast-google @connorcoley @n8kim1 PTAL since I had to guess at what was going on in some places.

File: editor/js/outcomes.js
Comment (connorcoley) [inline editor/js/outcomes.js:320]: I don't love the variable name `processes` here (and previous function names like `loadProcess`). Is it worth changing this to processedData, loadProcessedData, etc.?

File: editor/js/reaction.js
Comment (connorcoley) [inline editor/js/reaction.js:292]: It's more correct to say that this function returns a `Promise`. It seems like there is a lot of discussion around how to document type when it is fulfilled or rejected, but no clear consensus. Google's guide seems to suggest you would want to use `!Promise<!Uint8Array>` here

File: editor/js/reaction.js
Comment (connorcoley) [inline editor/js/reaction.js:326]: Same comment re: `!Promise<!Uint8Array>`

File: editor/js/reaction.js
Comment (connorcoley) [inline editor/js/reaction.js:481]: Could add "The new copy of the template" to the return documentation

File: editor/js/reaction.js
Comment (connorcoley) [inline editor/js/reaction.js:832]: Is this the correct way to record the type, or should it be `{?typeof jspb.Message}`?

File: editor/js/conditions.js
Comment (skearnes) [reply editor/js/conditions.js:121]: Done for all instances.

File: editor/js/dataset.js
Comment (skearnes) [reply editor/js/dataset.js:106]: Done.

File: editor/js/dataset.js
Comment (skearnes) [reply editor/js/dataset.js:108]: Removed; I didn't need to add this and the linter doesn't require it.

File: editor/js/outcomes.js
Comment (skearnes) [reply editor/js/outcomes.js:320]: I'll do this as part of #349 so we only change JSDoc in this PR; sound good?

File: editor/js/reaction.js
Comment (skearnes) [reply editor/js/reaction.js:292]: Done.

File: editor/js/reaction.js
Comment (skearnes) [reply editor/js/reaction.js:481]: Done.

File: editor/js/reaction.js
Comment (skearnes) [reply editor/js/reaction.js:832]: Done.

**Comment by skearnes** [comment]
Thanks for the comments; PTAL.

File: editor/js/outcomes.js
Comment (connorcoley) [reply editor/js/outcomes.js:320]: Yep, sounds good

**Comment by connorcoley** [comment]
LGTM. I can't approve the PR since I'm also an author. Would be good to get another set of eyes on it

File: editor/js/codes.js
Comment (n8kim1) [inline editor/js/codes.js:97]: Is it worth noting which single code this is? (we do describe this in the node param description, so it's not necessary)

**Comment by antonkast-google** [comment]
I will make a pass later today and see if I can add anything.

File: editor/js/compounds.js
Comment (n8kim1) [inline editor/js/compounds.js:409]: "submission of a molecular drawing" perhaps? Was a little confusing to me at first glance...

File: editor/js/inputs.js
Comment (n8kim1) [inline editor/js/inputs.js:47]: "Creates and populates"? since this method involves a call to `add`, as opposed to loadInputUnnamed which doesn't actually add a new div

**Review by n8kim1 — APPROVED** [review]

**Comment by n8kim1** [comment]
Looks good to me (especially annotations); I just have a few small suggestions about descriptions (that I don't feel too strongly about anyways). Would still like Anton's approval too.

Thanks Steve and Connor!

File: editor/js/codes.js
Comment (skearnes) [reply editor/js/codes.js:97]: Yeah, I think we could add "specified by `node`" but that feels redundant. Do you agree?

File: editor/js/compounds.js
Comment (skearnes) [reply editor/js/compounds.js:409]: Done.

File: editor/js/inputs.js
Comment (skearnes) [reply editor/js/inputs.js:47]: That's fair; I can update this in many places.

Comment (skearnes) [reply editor/js/inputs.js]: Done in 746da3e5377414e24279e939c62e2d70df7fc0ef

File: editor/js/codes.js
Comment (n8kim1) [reply editor/js/codes.js:97]: Yep that's fair -- as-is is good with me.

**Review by antonkast-google — APPROVED** [review]

**Comment by antonkast-google** [comment]
Nothing to add. LGTM

**Pull request merged**
